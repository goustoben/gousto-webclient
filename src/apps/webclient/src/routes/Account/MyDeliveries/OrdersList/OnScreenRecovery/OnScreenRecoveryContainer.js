import { connect } from 'react-redux'

import { getRecoveryContent } from 'actions/onScreenRecovery'
import { getIsMultiSkipEnabled } from 'selectors/features'
import { getHasBoxesToSkip } from 'selectors/user'
import { OnScreenRecovery } from './OnScreenRecovery'

const mapStateToProps = (state) => ({
  triggered: state.onScreenRecovery.get('triggered'),
  visible: state.onScreenRecovery.get('modalVisibility'),
  isMultiSkipEnabled: getIsMultiSkipEnabled(state),
  hasBoxesToSkip: getHasBoxesToSkip(state)
})

const mapDispatchToProps = {
  getRecoveryContent,
}

export const OnScreenRecoveryContainer = connect(mapStateToProps, mapDispatchToProps)(OnScreenRecovery)
