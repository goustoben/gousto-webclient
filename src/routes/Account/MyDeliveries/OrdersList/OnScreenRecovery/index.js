export { OnScreenRecoveryContainer as OnScreenRecovery } from './OnScreenRecoveryContainer'
