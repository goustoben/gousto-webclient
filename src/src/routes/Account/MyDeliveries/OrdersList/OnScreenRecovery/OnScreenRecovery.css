@value HaricotLight, Haricot from 'styles/colors.css';
@value BreakpointToMedium from 'design-language/breakpoints.css';

@value borderRadius 3px;

.container {
	width: 560px;
	text-align: left;
	height: 100%;
	display: grid;
	grid-template-columns: 1fr;
	grid-template-rows: auto auto 1fr auto;
	grid-template-areas:
		"backgroundHeader"
		"header"
		"discount"
		"fixedToBottom";

	@media BreakpointToMedium {
		width: auto;
	}
}

.backgroundHeader {
	grid-area: backgroundHeader;
	background-image: url('media/images/pause-osr-modal-bg.png');
	background-size: cover;
	background-position: center center;
	background-repeat: no-repeat; 
	height: 200px;
}

.header {
	grid-area: header;
	padding-left: 1rem;
}

.closeIcon {
	background: url('media/svgs/icon-X.svg') center center no-repeat;
	background-color: white;
	padding: 1.5rem;
	width: 20px;
	height: 20px;
	display: block;
	cursor: pointer;
}

.bodyContainer {
	padding: 0 2rem 2rem 2rem;
}

.subHeader {
	composes: fontStyleSubHead from 'design-language/typography.css';
	margin: 0 0 0 1rem;
}

.discountDetails {
	grid-area: discount;
	padding: 1rem;
	background-color: HaricotLight;
	border-radius: borderRadius;
	margin: 2rem 0 0;
	display: flex;

	.discountIconContainer {
		background-color: Haricot;
		border-radius: borderRadius;
		padding: 1.5rem;
		height: 65px;

		.discountIcon {
			width: 35px;
			height: 35px;
		}
	}
}

.ctaContainer {
	text-align: center;
	padding: 1rem 2rem 2rem 2rem;
	grid-area: fixedToBottom;

	button:first-child {
		margin-bottom: 1rem;
	}

	@media BreakpointToMedium {
		padding: 1rem 1.5rem 1.5rem 1.5rem;
	}
}

