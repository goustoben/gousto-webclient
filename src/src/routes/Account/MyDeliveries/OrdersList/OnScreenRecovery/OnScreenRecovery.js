import PropTypes from 'prop-types'
import React from 'react'

import ModalComponent, { ModalContent, ModalTitle } from 'ModalComponent'

import css from './OrderSkipRecovery.css'

import Title from './Title'
import Offer from './Offer'
import ValueProposition from './ValueProposition'
import Header from './Header'
import Footer from './Footer'

const propTypes = {
  visible: PropTypes.bool,
  orderId: PropTypes.string,
  deliveryDayId: PropTypes.string,
  orderType: PropTypes.string,
  keepOrder: PropTypes.func.isRequired,
  cancelPendingOrder: PropTypes.func.isRequired,
  cancelProjectedOrder: PropTypes.func.isRequired,
  title: PropTypes.string,
  offer: PropTypes.object,
  valueProposition: PropTypes.shape({
    message: PropTypes.string,
    title: PropTypes.string,
  }),
  callToActions: PropTypes.shape({
    confirm: PropTypes.string,
    keep: PropTypes.string,
  }),
  triggered: PropTypes.bool,
  orderDate: PropTypes.string,
  getSkipRecoveryContent: PropTypes.func,
}

class OrderSkipRecovery extends React.PureComponent {

  componentDidUpdate(prevProps) {
    const { triggered, orderId, orderDate, deliveryDayId, orderType, getSkipRecoveryContent } = this.props
    const actionTriggered = (orderType === 'pending') ? 'Cancel' : 'Skip'

    if (triggered && (prevProps.triggered !== triggered)) {

      getSkipRecoveryContent({
        orderId,
        orderDate,
        deliveryDayId,
        status: orderType,
        actionTriggered,
      })
    }
  }

  skipCancelOrder(orderId, deliveryDayId, orderType, cancelPendingOrder, cancelProjectedOrder) {
    if (orderType === 'pending') {
      cancelPendingOrder(orderId, deliveryDayId)
    } else {
      cancelProjectedOrder(deliveryDayId)
    }
  }

  render() {
    const { visible, deliveryDayId, orderId, orderType, keepOrder, cancelPendingOrder, cancelProjectedOrder, title, offer, valueProposition, callToActions } = this.props
    const onClickKeepOrder = () => keepOrder({ orderId, deliveryDayId, status: orderType })
    const onClickSkipCancel = () => this.skipCancelOrder(orderId, deliveryDayId, orderType, cancelPendingOrder, cancelProjectedOrder)

    return (
      <ModalComponent visible={visible}>
        <Header offer={offer} />
        <div className={css.container}>
          <ModalTitle>
            <Title title={title} orderType={orderType} />
          </ModalTitle>
          <ModalContent>
            <Offer offer={offer} />
            {(offer && valueProposition) ? <hr className={css.rule} /> : null}
            <ValueProposition valueProposition={valueProposition} />
     </ModalContent>
          <Footer orderType={orderType} callToActions={callToActions} onClickKeepOrder={onClickKeepOrder} onClickSkipCancel={onClickSkipCancel} />
        </div>
      </ModalComponent>
    )
  }
}

OrderSkipRecovery.propTypes = propTypes

export { OrderSkipRecovery }
