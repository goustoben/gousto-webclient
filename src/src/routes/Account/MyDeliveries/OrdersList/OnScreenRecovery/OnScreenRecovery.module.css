.modalComponent {
	@media (max-width: 420px) {
		width: 320px;
		transform: none;
	}
}
