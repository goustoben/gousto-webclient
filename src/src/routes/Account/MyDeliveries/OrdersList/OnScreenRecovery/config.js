export const ORDER_TYPE = 'order'
export const SUBSCRIPTION_TYPE = 'subscription'
