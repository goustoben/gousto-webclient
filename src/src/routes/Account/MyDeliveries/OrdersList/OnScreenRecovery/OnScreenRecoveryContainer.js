import { connect } from 'react-redux'

import { getSkipRecoveryContent, keepOrder, cancelPendingOrder, cancelProjectedOrder } from 'actions/onScreenRecovery'

import { OnScreenRecovery } from './OnScreenRecovery'

const mapStateToProps = (state) => ({
  triggered: state.onScreenRecovery.get('triggered'),
  visible: state.onScreenRecovery.get('modalVisibility'),
  orderId: state.onScreenRecovery.get('orderId'),
  orderDate: state.onScreenRecovery.get('orderDate'),
  deliveryDayId: state.onScreenRecovery.get('deliveryDayId'),
  orderType: state.onScreenRecovery.get('orderType'),
  boxNumber: state.subscription.getIn(['subscription', 'currentBoxNumber']),
  title: state.onScreenRecovery.get('title'),
  offer: state.onScreenRecovery.get('offer'),
  valueProposition: state.onScreenRecovery.get('valueProposition'),
  callToActions: state.onScreenRecovery.get('callToActions'),
})

const mapDispatchToProps = {
  getSkipRecoveryContent,
  keepOrder,
  cancelPendingOrder,
  cancelProjectedOrder,

}

export const OnScreenRecoveryContainer = connect(mapStateToProps, mapDispatchToProps)(OnScreenRecovery)
