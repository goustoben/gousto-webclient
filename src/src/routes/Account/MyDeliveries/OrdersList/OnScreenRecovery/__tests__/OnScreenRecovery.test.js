import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { OnScreenRecovery } from '../OnScreenRecovery'

jest.mock('components/Overlay', () => 'Overlay')

describe('Order Skip Recovery Modal', () => {
  let wrapper
  const keepOrder = jest.fn()
  const cancelPendingOrder = jest.fn()

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('Initial Render', () => {
    const valueProposition = {
      title: 'value proposition title',
      message: 'value proposition message',
    }

    beforeAll(() => {
      wrapper = shallow(
        <OnScreenRecovery
          title="modal title"
          orderType="pending"
          onKeep={keepOrder}
          onConfirm={cancelPendingOrder}
          featureFlag
          valueProposition={valueProposition}
          keepCopy="keep"
          confirmCopy="confirm"
        />
      )
    })

    test('should render snapshot', () => {
      const tree = renderer.create(
        <OnScreenRecovery
          title="modal title"
          orderType="pending"
          onKeep={keepOrder}
          onConfirm={cancelPendingOrder}
          featureFlag
          valueProposition={valueProposition}
          keepCopy="keep"
          confirmCopy="confirm"
        />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('should display modal title', () => {
      const modalTitle = wrapper.find('Title')

      expect(modalTitle.length).toBe(1)
    })

    test('should pass correct props to title', () => {
      const modalTitle = wrapper.find('Title')

      expect(modalTitle.props().title).toBe('modal title')
    })

    test('should display modal value proposition', () => {
      const modalValueProposition = wrapper.find('ValueProposition')

      expect(modalValueProposition.length).toBe(1)
    })

    test('should pass correct props to value proposition', () => {
      const modalValueProposition = wrapper.find('ValueProposition')

      expect(modalValueProposition.props().valueProposition).toEqual(valueProposition)
    })

    test('should display modal footer', () => {
      const footer = wrapper.find('Footer')

      expect(footer.length).toBe(1)
    })

    test('should pass correct props to footer', () => {
      const footer = wrapper.find('Footer')

      expect(footer.props().keepCopy).toBe('keep')
      expect(footer.props().confirmCopy).toEqual('confirm')
      expect(typeof footer.props().onKeep).toBe('function')
      expect(typeof footer.props().onConfirm).toBe('function')
    })
  })

  describe('Alternative Render', () => {
    test('should call getRecoveryContent', () => {
      const getRecoveryContent = jest.fn()
      wrapper = shallow(
        <OnScreenRecovery
          orderId="14245"
          triggered={false}
          orderDate="2018-09-24T13:27:09.487Z"
          deliveryDayId="23001"
          orderType="pending"
          getRecoveryContent={getRecoveryContent}
        />
      )

      const prevProps = wrapper.props()

      wrapper.setProps({
        triggered: true,
      })

      wrapper.instance().componentDidUpdate(prevProps)

      expect(getRecoveryContent).toHaveBeenCalled()
    })
  })
})
