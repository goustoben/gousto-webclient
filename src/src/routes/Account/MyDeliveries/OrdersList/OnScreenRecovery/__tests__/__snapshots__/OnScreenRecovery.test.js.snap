// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Recovery Modal Initial Render should render snapshot 1`] = `
<Overlay
  contentClassName=""
  from="top"
  onBackgroundClick={null}
  open={false}
>
  <div
    className="modalComponent modalComponent"
  >
    <div
      className="spacer"
    />
    <div
      className="container"
    >
      <div
        className="modalTitle"
      >
        <div
          className="title"
        >
          modal title
        </div>
      </div>
      <div
        className="modalContent"
      >
        <div
          className="container"
        >
          <div
            className="title"
          >
            value proposition title
          </div>
          <div
            className="message"
          >
            value proposition message
          </div>
        </div>
      </div>
      <div
        className="modalFooter"
      >
        <div
          className="confirm"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          confirm
        </div>
        <button
          className="keep"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          keep
        </button>
      </div>
    </div>
  </div>
</Overlay>
`;
